test/unit/test_materialization.py

```diff
@@ -473,3 +474,21 @@ def test_close_stale_task_log(conn):
         ).fetchall()
         assert len(rows) == 1, f"Expected 1 row, got {rows}"
         assert rows[0]["QUERY_ID"] == new_query_id
+
+
+def test_migrate_qtag(conn, current_timezone):
+    with conn() as cnx, cnx.cursor(DictCursor) as cur:
+        sql = """
+UPDATE internal_reporting_mv.query_history_complete_and_daily{tbl_suffix}
+SET    qtag_filter = NULL
+where start_time::date = current_date
+"""
+        res = cur.execute(sql.format(tbl_suffix="")).fetchall()
+        assert len(res) == 1
+        updated_rows = res[0].get("number of rows updated")
+        assert updated_rows is not None
+
+        res = cur.execute("CALL internal.update_qtag_day()").fetchall()
+        assert len(res) == 1
+        assert len(res[0]) == 1
+        assert abs(res[0]["UPDATE_QTAG_DAY"] - updated_rows) < updated_rows * 0.2
```

Comment:
I'm struggling with what this row is asserting -- we're checking that one execution of this procedure is never updating more than 20% of the total number of rows from today?

Doesn't that make this dependent on the number of query history rows that have qtags? Feels brittle.

Reply:
Ok, I get it now -- verifying that the two values are very close together (i.e. we updated all of the rows we expected to update)